plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2398,6 +2414,99 @@ else if (isMetadataColumnId(columnHandle.getId())) {
                 false));
     }
 
+    @Override
+    public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggregation(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            List<AggregateFunction> aggregates,
+            Map<String, ColumnHandle> assignments,
+            List<List<ColumnHandle>> groupingSets)
+    {
+        IcebergTableHandle tableHandle = (IcebergTableHandle) handle;
+
+        // Iceberg's metadata cannot be used for aggregation calculation.
+        // As equality deletes do not reflect at the metadata/count level.
+        if (hasEqualityDeletes(session, tableHandle)) {
+            return Optional.empty();
+        }
+
+        if (!isAggregationPushdownEnabled(session)) {
+            return Optional.empty();
+        }
+
+        // not supporting unenforced predicate
+        if (!tableHandle.getUnenforcedPredicate().isNone()
+                && tableHandle.getUnenforcedPredicate().getDomains().isPresent()
+                && !tableHandle.getUnenforcedPredicate().getDomains().get().isEmpty()) {
+            return Optional.empty();
+        }
+
+        // not supporting group by
+        if (!groupingSets.isEmpty()
+                && groupingSets.get(0) != null
+                && groupingSets.get(0).size() > 0) {
```

Comment:
You're inspecting the first grouping set only. Why not inspect the second grouping set as well?

The correct way to check for global aggregation would be `groupingSets.equals(List.of(List.of()))`
but again, it would be worthwhile to support the GROUP BY

(note that PushAggregationIntoTableScan currently doesn't support GROUPING SETS, so supporting them in v1 doesn't make sense)